lang/java/reef-runtime-standalone/src/main/java/org/apache/reef/runtime/standalone/driver/ProcessContainer.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.standalone.driver;
+
+import com.jcraft.jsch.Session;
+import org.apache.reef.annotations.audience.Private;
+import org.apache.reef.annotations.audience.TaskSide;
+import org.apache.reef.runtime.common.files.REEFFileNames;
+import org.apache.reef.runtime.standalone.process.SshProcess;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A Container that runs an Evaluator in a Process.
+ */
+@Private
+@TaskSide
+final class ProcessContainer implements Container {
+
+  private static final Logger LOG = Logger.getLogger(ProcessContainer.class.getName());
+
+  private final String errorHandlerRID;
+  private final String nodeID;
+  private final File folder;
+  private final String containedID;
+  private final int megaBytes;
+  private final int numberOfCores;
+  private final REEFFileNames fileNames;
+  private final File reefFolder;
+  private final File localFolder;
+  private final File globalFolder;
+  private Thread theThread;
+  private SshProcess process;
+  private Session remoteSession;
+  private String remoteHostName;
+  private final String nodeFolder;
+
+  /**
+   * @param errorHandlerRID the remoteID of the error handler.
+   * @param nodeID          the ID of the (fake) node this Container is instantiated on
+   * @param containedID     the  ID used to identify this container uniquely
+   * @param folder          the folder in which logs etc. will be deposited
+   */
+  ProcessContainer(final String errorHandlerRID,
```

Comment:
Maybe rename this to `SshProcessContainer`? It seems that the main purpose of this class is to hold onto parameters required to create `SshProcess`. Also, quite a few methods of this class seem to be only used in `reef-runtime-local`. Considering these issues, could you please refactor this class?